tests/test_2024_10_08.py

```diff
@@ -0,0 +1,303 @@
+    assert not l1_token_bridge.isDepositsEnabled()
+
+    # LidoLocator has new implementation
+    assert (
+        lido_locator_proxy.proxy__getImplementation() == LIDO_LOCATOR_IMPL_NEW
+    ), "New LidoLocator address is incorrect"
+
+    # Multisig has been assigned as deposit enabler
+    assert l1_token_bridge.hasRole(DEPOSITS_ENABLER_ROLE, L1_EMERGENCY_BRAKES_MULTISIG)
+
+    interface.OptimismPortal2(L1_OPTIMISM_PORTAL_2)  # load OptimismPortal2 contract ABI to see events
+
+    evs = group_voting_events(vote_tx)
+
+    validate_contract_upgrade(evs[0], L1_OPTIMISM_TOKENS_BRIDGE_IMPL_NEW)
+    validate_contract_finalization(evs[1], 2)
+    validate_contract_upgrade(evs[2], LIDO_LOCATOR_IMPL_NEW)
+    validate_contract_role_granting(evs[3], DEPOSITS_ENABLER_ROLE, L1_EMERGENCY_BRAKES_MULTISIG, AGENT)
+    validate_optimism_upgrade_call(
+        evs[4],
+        L2_OPTIMISM_GOVERNANCE_EXECUTOR,
+        AGENT,
+        L2_OPTIMISM_TOKENS_BRIDGE,
+        L2_OPTIMISM_TOKENS_BRIDGE_IMPL_NEW,
+        L2_OPTIMISM_WSTETH_TOKEN,
+        L2_OPTIMISM_WSTETH_TOKEN_IMPL_NEW,
+    )
+
+
+def validate_contract_upgrade(events: EventDict, implementation: str):
+    _events_chain = ["LogScriptCall", "LogScriptCall", "Upgraded", "ScriptResult"]
+    validate_events_chain([e.name for e in events], _events_chain)
+    assert events.count("Upgraded") == 1
+    assert events["Upgraded"]["implementation"] == implementation, "Wrong implementation"
+
+
+def validate_contract_finalization(events: EventDict, version: int):
+    _events_chain = ["LogScriptCall", "LogScriptCall", "ContractVersionSet", "ScriptResult"]
+    validate_events_chain([e.name for e in events], _events_chain)
+    assert events.count("ContractVersionSet") == 1
+    assert events["ContractVersionSet"]["version"] == version
+
+
+def validate_contract_role_granting(events: EventDict, role: str, account: str, sender: str):
+    _events_chain = ["LogScriptCall", "LogScriptCall", "RoleGranted", "ScriptResult"]
+    validate_events_chain([e.name for e in events], _events_chain)
+    assert events.count("RoleGranted") == 1
+    assert events["RoleGranted"]["role"] == role
+    assert events["RoleGranted"]["account"] == account
+    assert events["RoleGranted"]["sender"] == sender
+
+
+def validate_optimism_upgrade_call(
+    events: EventDict,
+    target: str,
+    sender: str,
+    l2_bridge_proxy: str,
+    l2_bridge_impl: str,
+    l2_wst_proxy: str,
+    l2_wst_impl: str,
+):
+    _events_chain = [
+        "LogScriptCall",
+        "LogScriptCall",
+        "TransactionDeposited",
+        "SentMessage",
+        "SentMessageExtension1",
+        "ScriptResult",
+    ]
+    validate_events_chain([e.name for e in events], _events_chain)
+    assert events.count("SentMessage") == 1
+    # Check bytecode that was sent to messenger to update L2 bridge and wstETH token
+    encoded_l2_upgrade_call = encode_l2_upgrade_call(l2_bridge_proxy, l2_bridge_impl, l2_wst_proxy, l2_wst_impl)
```

Comment:
Decoding the emitted "message" and checking that it contains the correct values will be more reliable.

Reply:
Will unparse to check params